docs/content/docs/dev/table/materialized-table/overview.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Overview
+weight: 1
+type: docs
+aliases:
+- /dev/table/materialized-table.html
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Introduction
```

Comment:
I think we should split the materialized table concept and statement into two page. The overview page introduce the core concepts,  statement page introduce all supported statements. So I think the docs catalog structure can be designed as follows:

overview: Introduce the core concepts
# Introduction

# Core Concepts

## Schema
   Same as regular table

## Freshness
   data freshness

## Refresh Mode
   Refresh job mode: Streamig or batch

## Definition Query
    Data product logic


statements: Introduce statements
Materialized Table supports the following statements for now:
    CREATE MATERIALIZED TABLE
    ALTER MATERIALIZED TABLE
    DROP MATERIALIZED TABLE

# CREATE MATERIALIZED TABLE
    syntax template
## PRIMARY KEY
## PARTITIONED BY
## WITH Options
## FRESHNESS
## REFRESH_MODE
## AS <select_statment>
    Does not need to introduce the implementation mechanism, just state that the normal query statement is supported,  see https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/overview/
## LIMITATION:
    Does not support explicitly specifying columns.
    Does not support modifying the select statement after the materialized table is created. 
    Does not support referencing temporary tables, views, or functions in Select statements currently.
## Example
   give some different pattern examples

# ALTER MATERIALIZED TABLE
    syntax template
## SUSPEND
    introduce function & example
## RESUME
    introduce function example
## REFRESH
    introduce function example

# DROP MATERIALIZED TABLE
    syntax template
    example


# LIMITATION
    only SQL Gateway support, java & scala & python api doesn't support
    depend on catalog & catalog store
    full mode depend on workflow scheduler